Hardware/Hardware/CardReader/StripeCardReader/StripeCardReaderService.swift

```diff
@@ -499,10 +499,10 @@ extension StripeCardReaderService: BluetoothReaderDelegate {
 
     public func reader(_ reader: Reader, didFinishInstallingUpdate update: ReaderSoftwareUpdate?, error: Error?) {
         if let error = error {
-            let underlyingError = UnderlyingError(with: error)
-            if underlyingError != .commandCancelled {
-                softwareUpdateSubject.send(.failed(error: error))
-            }
+            softwareUpdateSubject.send(.failed(
+                error: CardReaderServiceError.softwareUpdate(underlyingError: UnderlyingError(with: error),
+                                                             batteryLevel: reader.batteryLevel?.doubleValue))
+            )
```

Comment:
The logic for ignoring cancellations has moved to WooCommerce/Classes/ViewRelated/Dashboard/Settings/CardReadersV2/CardReaderSettingsConnectedViewModel.swift

The pre-existing logic of using .commandCancelled to ignore cancelations appears to be incorrect, as the reader sends `.readerSoftwareUpdateFailedInterrupted` in that case.

Reply:
Good catch!